File: rust_with_atcoder/atcoder_practice/abc051_b/examples/051b.ts
import * as fs from "fs";

const [k, s] = fs.readFileSync(0, "utf8").trim().split(/\s+/).map(Number);

let count = 0;

for (let x = 0; x <= k; x++) {
	for (let y = 0; y <= k; y++) {
		const z = s - x - y;
		if (0 <= z && z <= k) {
			count += 1;
		}
	}
}

console.log(count);

File: rust_with_atcoder/atcoder_practice/abc051_b/examples/051b.py
import sys


def main() -> None:
    k, s = map(int, sys.stdin.read().split())

    count = 0

    for x in range(k + 1):
        for y in range(k + 1):
            z = s - x - y
            if 0 <= z <= k:
                count += 1

    print(count)


if __name__ == "__main__":
    main()
